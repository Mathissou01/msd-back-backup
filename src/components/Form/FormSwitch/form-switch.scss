@use "src/styles/01-settings/settings.colors" as colors;

.c-FormSwitch {
  &__Input[type="checkbox"] {
    width: 0;
    height: 0;
    visibility: hidden;
  }

  &__Label {
    position: relative;
    display: block;
    width: 40px;
    height: 20px;
    text-indent: -9999px;
    background: colors.$expert-blue;
    border-radius: 10px;
    cursor: pointer;
  }

  &__Label::after {
    position: absolute;
    top: 1px;
    left: 1px;
    width: 18px;
    height: 18px;
    background: colors.$white;
    border-radius: 50%;
    content: "";
  }

  &__Input:checked + &__Label {
    background: colors.$graph-low;
  }

  &__Input:checked + &__Label::after {
    left: calc(100% - 1px);
    transform: translateX(-100%);
  }
}
