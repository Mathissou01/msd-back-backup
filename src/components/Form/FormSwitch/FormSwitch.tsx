import { useFormContext } from "react-hook-form";
import "./form-switch.scss";

interface IFormSwitchProps {
  name: string;
  isDisabled?: boolean;
}

export default function FormSwitch({ name, isDisabled }: IFormSwitchProps) {
  /* Local Data */
  const { register } = useFormContext();

  return (
    <>
      <input
        className="c-FormSwitch__Input"
        type="checkbox"
        id={name}
        {...register(name)}
        data-testid="form-switch"
        disabled={isDisabled}
      />
      <label htmlFor={name} className="c-FormSwitch__Label">
        {name}
      </label>
    </>
  );
}
