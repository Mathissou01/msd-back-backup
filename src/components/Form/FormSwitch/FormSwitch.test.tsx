import React, { ReactNode } from "react";
import { FormProvider, useForm } from "react-hook-form";
import { fireEvent, render, screen } from "@testing-library/react";
import FormSwitch from "./FormSwitch";

const mock = {
  name: "form-switch",
};

const Wrapper = (props: { children: ReactNode }) => {
  const formMethods = useForm();
  return <FormProvider {...formMethods}>{props.children}</FormProvider>;
};

describe("FormSwitch", () => {
  it("renders", () => {
    const { container } = render(
      <Wrapper>
        <FormSwitch name={mock.name} />
      </Wrapper>,
    );

    const formSwitch = screen.getByTestId("form-switch");
    expect(formSwitch).toBeInTheDocument();
    expect(formSwitch).not.toHaveAttribute("disabled");

    expect(container).toMatchSnapshot();
  });

  it("when checked", async () => {
    render(
      <Wrapper>
        <FormSwitch name={mock.name} />
      </Wrapper>,
    );

    const formSwitch = screen.getByTestId("form-switch");
    fireEvent.click(formSwitch);
    expect(formSwitch).toBeChecked();
  });

  it("when unchecked", async () => {
    render(
      <Wrapper>
        <FormSwitch name={mock.name} />
      </Wrapper>,
    );

    const formSwitch = screen.getByTestId("form-switch");
    fireEvent.doubleClick(formSwitch);
    expect(formSwitch).not.toBeChecked();
  });

  it("can be disabled", () => {
    render(
      <Wrapper>
        <FormSwitch name={mock.name} isDisabled />
      </Wrapper>,
    );

    const formSwitch = screen.getByTestId("form-switch");
    expect(formSwitch).toHaveAttribute("disabled");
    fireEvent.click(formSwitch);
  });
});
