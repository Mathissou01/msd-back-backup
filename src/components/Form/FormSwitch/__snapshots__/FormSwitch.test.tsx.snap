// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSwitch renders 1`] = `
<div>
  <input
    class="c-FormSwitch__Input"
    data-testid="form-switch"
    id="form-switch"
    name="form-switch"
    type="checkbox"
  />
  <label
    class="c-FormSwitch__Label"
    for="form-switch"
  >
    form-switch
  </label>
</div>
`;
